apps/orders/models.py

```diff
@@ -1,45 +1,105 @@
 from django.contrib.auth.base_user import AbstractBaseUser
 from django.core.validators import MinValueValidator, MaxValueValidator
-from django.db import models
+from django.db import models, IntegrityError
+from django.db.models import Sum, F
+from django.db.models.signals import pre_save
+from django.dispatch import receiver
+from rest_framework.exceptions import ValidationError
 
 from apps.products.models import Products
 from apps.users.models import User, UserAddress
-
+from apps.common.models import BaseModel
 
 # Create your models here.
 
-class BaseModelMixin(models.Model):
-    created_at = models.DateTimeField(auto_now_add=True)
-    updated_at = models.DateTimeField(auto_now=True)
+
+class Cart(BaseModel):
+    user = models.ForeignKey(User, on_delete=models.CASCADE, related_name='carts')
+    is_archived = models.BooleanField(default=False)
 
     class Meta:
-        abstract = True
+        ordering = ['-id']
 
+    def add_item(self, item_id, count=None):
+        try:
+            item, was_created = self.items.get_or_create(product_id=item_id)
+        except IntegrityError:
+            raise ValidationError({'product': 'Invalid product'})
 
-class Cart(BaseModelMixin):
-    user = models.ForeignKey(User, on_delete=models.CASCADE, related_name='cart')
-    is_archived = models.BooleanField(default=False)
+        if count < 1:
+            raise ValidationError({'count': 'Item count can\'t be less than 1'})
+
+        item.count = count
+        item.price = item.product.price
+        item.discount = item.product.discount
+
+        item.save()
```

Comment:
faci un request ca sa creezi itemul si apoi ii faci inca un save. Prea multe requesturi in baza, incearca sa faci cu un sigur request